qiskit/qobj/qobj.py

```diff
@@ -189,24 +183,54 @@ def __init__(self, instructions, **kwargs):
 
 
 class QobjExperimentHeader(QobjItem):
-    """Header for a Qobj.
+    """Header for a Qobj Experiment .
 
     Attributes defined in the schema but not required:
         name (str): experiment name.
     """
     pass
 
 
+class QobjExperimentConfig(QobjItem):
+    """Config for a Qobj Experiment.
+
+    """
+    pass
+
+
 class QobjInstruction(QobjItem):
     """Quantum Instruction.
 
     Attributes:
-        name(str): name of the gate.
-        qubits(list): list of qubits to apply to the gate.
+        name(str): name of the operation.
+    Optional Attributes:
+        qubits(list): list of qubits to apply to the operation.
+        params(list): list of params for the operation.
+        memory(list): list of memory to apply to the operation.
+        conditional(QobjConditional): conditional Qobj
     """
     REQUIRED_ARGS = ['name']
 
     def __init__(self, name, **kwargs):
         self.name = name
 
         super().__init__(**kwargs)
+
+
+class QobjConditional(QobjItem):
+    """Quantum Conditional.
+
+    Attributes:
+        mask (hex): mask of the conditional
+        type (string): type of the conditional
+        val (hex): value of the conditional
+    """
+    # pylint: disable=redefined-builtin
+    REQUIRED_ARGS = ['mask', 'type', 'val']
+
+    def __init__(self, mask, type, val):
```

Comment:
For consistency with the rest of the Qobj items, can we keep the `**kwargs` argument to `__init__` (passing it to the `super()` call afterwards)? It will not really make a big difference in the regular use cases, but it would technically keep the `QobjConditional` in the spirit of the "extendable schema", being able to append attributes to that structure as needed.

Reply:
But this one is not extendable

Reply:
Hmm I think it depends on the point of view (and how the specs are interpreted) - technically, every entity was meant to be extendable (in the sense that it can contain additional fields that are not defined formally). Not really a PR-blocking issue (specially if we aim to move the implementation of Qobj to another mechanism soon-ish!).

Reply:
yeah, i a agree for the other objects in qobj as they are but the conditionals are well defined and can not be extended. Saying this is the old style conditionals and will change in a followup pr.

Reply:
:+1: